api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/dropwizard/RESTMetrics.java

```diff
@@ -16,118 +16,122 @@
-        add(RESTMetaData.forRead(HTTPMethod.GET, "counters/id/stats/tags/tags"));
-
-        add(RESTMetaData.forWrite(HTTPMethod.POST, "metrics"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "metrics"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "metrics/tags"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "metrics/tags/tags"));
-        add(RESTMetaData.forWrite(HTTPMethod.POST, "metrics/raw"));
-        add(RESTMetaData.forRead(HTTPMethod.POST, "metrics/stats/query"));
-        add(RESTMetaData.forRead(HTTPMethod.POST, "metrics/stats/batch/query"));
-
-        // TODO Should we make string and availability metrics optional since they are not used in openshift?
-
-        add(RESTMetaData.forWrite(HTTPMethod.POST, "strings"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "strings"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "strings/id"));
-        add(RESTMetaData.forWrite(HTTPMethod.DELETE, "strings/id"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "strings/tags/tags"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "strings/id/tags"));
-        add(RESTMetaData.forWrite(HTTPMethod.PUT, "strings/id/tags"));
-        add(RESTMetaData.forWrite(HTTPMethod.DELETE, "strings/tags/tags"));
-        add(RESTMetaData.forWrite(HTTPMethod.POST, "strings/id/raw"));
-        add(RESTMetaData.forWrite(HTTPMethod.POST, "strings/raw"));
-        add(RESTMetaData.forRead(HTTPMethod.POST, "strings/raw/query"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "strings/id/raw"));
-
-        add(RESTMetaData.forWrite(HTTPMethod.POST, "availability"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "availability"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "availability/id"));
-        add(RESTMetaData.forWrite(HTTPMethod.DELETE, "availability/id"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "availability/tags/tags"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "availability/id/tags"));
-        add(RESTMetaData.forWrite(HTTPMethod.PUT, "availability/id/tags"));
-        add(RESTMetaData.forWrite(HTTPMethod.DELETE, "availability/id/tags/tags"));
-        add(RESTMetaData.forWrite(HTTPMethod.POST, "availability/id/raw"));
-        add(RESTMetaData.forWrite(HTTPMethod.POST, "availability/raw"));
-        add(RESTMetaData.forRead(HTTPMethod.POST, "availability/raw/query"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "availability/id/raw"));
-        add(RESTMetaData.forRead(HTTPMethod.GET, "availability/id/stats"));
+    private MetricNameService metricNameService;
+
+    public void setMetricNameService(MetricNameService metricNameService) {
+        this.metricNameService = metricNameService;
+    }
+
+    public void initMetrics() {
+        String hostname = metricNameService.getHostName();
+
+        register(RESTMetaData.forRead(HTTPMethod.GET, "status", hostname));
+
+        register(RESTMetaData.forWrite(HTTPMethod.POST, "gauges", hostname));
```

Comment:
Yeah, these need to be dynamically generated. The endpoints are not going to change for the next 0.27.x release which is main concern right now. I would prefer to do this in a follow ticket because I need to get this merged asap.